src/mlpack/core/data/imputation_methods/mean_imputation.hpp

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file mean_imputation.hpp
+ * @author Keon Kim
+ *
+ * Definition and Implementation of the MeanImputation class.
+ */
+#ifndef MLPACK_CORE_DATA_IMPUTE_STRATEGIES_MEAN_IMPUTATION_HPP
+#define MLPACK_CORE_DATA_IMPUTE_STRATEGIES_MEAN_IMPUTATION_HPP
+
+#include <mlpack/core.hpp>
+
+using namespace std;
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A simple mean imputation
+ */
+template <typename T>
+class MeanImputation
+{
+ public:
+  void Apply (const arma::Mat<T>& input,
+              arma::Mat<T>& output,
+              const T& mappedValue,
+              const size_t dimension,
+              const bool transpose = true)
+  {
+    // initiate output
+    output = input;
+
+    double sum;
+    size_t elems = 0; // excluding nan or missing target
+
+    using PairType = std::pair<size_t, size_t>;
+    // dimensions and indexes are saved as pairs inside this vector.
+    std::vector<PairType> targets;
+
+    // calculate number of elements and sum of them excluding mapped value or
+    // nan. while doing that, remember where mappedValue or NaN exists.
+    if (transpose)
+    {
+      for (size_t i = 0; i < input.n_rows; ++i)
+      {
+        if (input(i, dimension) == mappedValue)
+        {
+          targets.push_back(std::make_pair(i, dimension));
```

Comment:
Another approach(maybe you already know)--targets.emplace_back(i, dimension);

Different with push_back is emplace_back will construct the value directly,it may(or may not) provide better performance. Pick the one you like :)